packages/tools/client-debugger/client-debugger/src/messaging/AudienceMessages.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { IClient } from "@fluidframework/protocol-definitions";
+import { AudienceChangeLogEntry } from "../Logs";
+import { HasContainerId } from "./DebuggerMessages";
+
+import { IDebuggerMessage } from "./Messages";
+
+/**
+ * Metadata of clients within the Audience.
+ *
+ * @public
+ */
+export interface AudienceClientMetaData {
+	clientId: string;
+	client: IClient;
+}
+
+/**
+ * Inbound message requesting audience data from the Container with the specific ID.
+ * Will result in the {@link AudienceEventMessage} message being posted.
+ *
+ * @public
+ */
+export interface GetAudienceMessage extends IDebuggerMessage<HasContainerId> {
+	type: "GET_AUDIENCE";
+}
+
+/**
+ * Message data format used by {@link AudienceEventMessage}.
+ *
+ * @public
+ */
+export interface AudienceEventMessageData extends HasContainerId {
+	/**
+	 * Id of the client connected to the container
+	 */
+	clientId: string | undefined;
+	/**
+	 * Metadata of the current Audience state.
+	 */
+	audienceState: AudienceClientMetaData[];
+	/**
+	 * Connection history of members to the container
+	 */
+	audienceHistory: readonly AudienceChangeLogEntry[];
+}
+
+/**
+ * Audience event message which contains {@link AudienceEventMessageData} data
+ *
+ * @public
+ */
+export interface AudienceEventMessage extends IDebuggerMessage<AudienceEventMessageData> {
```

Comment:
I wonder if we should just call this `AudienceMessage`. Or `AudienceSummaryMessage` maybe? This doesn't really describe an "event".

We can always change this later, though. So feel free to close this with no action :)

Reply:
Changed to `AudienceSummaryMessageData` and `AudienceSummaryMessage ` for consistency.